src/middleware/checkDuplicates.ts

```diff
@@ -0,0 +1,35 @@
+import { NextFunction, Request, Response } from "express";
+import { databaseModel } from "../model";
+import { STATUS_CODES } from "../utils";
+
+export const checkDuplicates = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const { originalUrl, customName } = req.body;
+
+    if (customName) {
+      const checkName = await databaseModel.findOne({ shortUrl: customName });
```

Comment:
Instead of calling findOne twice, you can use the $or logical operator. Something like this

```
const duplicate = await databaseModel.findOne({
      $or: [
        { shortUrl: customName },
        { originalUrl: originalUrl },
      ],
    });
```